### PR TITLE
Order and line status verbose names

### LINKED ISSUE #1232: Order status and line status can't be translated
The status pipelines relies on strings to work. I just marked them translatable, but got an import error for that file - I presume marked strings can't be used as dictionary keys. That makes it hard to translate the order status and line status.

### PR DESCRIPTION
**Pull request by sasha0**
Fixes #1232

### CODE REVIEW

**Comment by codeinthehole** [comment]
First of all - thanks for all your excellent hard work.

Forgive my OCD, but I'd prefer you to write commit messages as per this guide: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

Eg:

- Keep the first line short and start it with a capital letter
- Use the present tense rather than the past tense

You don't need to include the number of the issue you're fixing in every commit message - better to just put "Fixes #..." in the PR description. It would be useful to have more informative titles and descriptions in the PRs too.  

I'll look at the code now...

File: src/oscar/apps/order/abstract_models.py [hunk 1]
```diff
@@ -134,6 +134,13 @@ def set_status(self, new_status):
     set_status.alters_data = True
 
     @property
+    def status_verbose_name(self):
+        status_verbose_names = getattr(settings, 'OSCAR_LINE_STATUS_VERBOSE_NAMES', {})
+        if self.status and status_verbose_names and self.status in status_verbose_names:
```
Comment (codeinthehole) [inline src/oscar/apps/order/abstract_models.py]: You don't really need the `.. and status_verbose_names ..` do you? The condition wouldn't error if `status_verbose_names` was an empty dict.

File: src/oscar/apps/order/abstract_models.py [hunk 2]
```diff
@@ -134,6 +134,13 @@ def set_status(self, new_status):
     set_status.alters_data = True
 
     @property
+    def status_verbose_name(self):
+        status_verbose_names = getattr(settings, 'OSCAR_LINE_STATUS_VERBOSE_NAMES', {})
```
Comment (codeinthehole) [inline src/oscar/apps/order/abstract_models.py]: This is the wrong setting to use isn't it? Should be `OSCAR_ORDER_STATUS_VERBOSE_NAMES`.

File: src/oscar/apps/order/abstract_models.py [hunk 3]
```diff
@@ -543,6 +550,13 @@ def set_status(self, new_status):
     set_status.alters_data = True
 
     @property
+    def status_verbose_name(self):
+        status_verbose_names = getattr(settings, 'OSCAR_ORDER_STATUS_VERBOSE_NAMES', {})
```
Comment (codeinthehole) [inline src/oscar/apps/order/abstract_models.py]: Wrong setting I think

**Comment by codeinthehole** [comment]
Couple of things to correct - but we should be able to squeeze this into 1.1

File: src/oscar/apps/order/abstract_models.py (hunk 2 above)
Comment (sasha0) [inline src/oscar/apps/order/abstract_models.py]: Intended to allow different settings for orders and lines. What would you suggest, one settings for both?

File: src/oscar/apps/order/abstract_models.py (hunk 1 above)
Comment (sasha0) [inline src/oscar/apps/order/abstract_models.py]: @codeinthehole fixed.

**Comment by sasha0** [comment]
@mvantellingen per our latest discussion - do we want to rework status pipeline on top of `django-fsm`?

**Comment by sasha0** [comment]
Per latest discussion with @mvantellingen, we'll experiment with `django-fsm` in order to rework statuses pipeline in the next versions. Thus, closing this PR since none of this changes will be used in the refactoring mentioned above.

**Pull request closed without merging**